src/pyhf/optimize/opt_tflow.py

```diff
@@ -1,76 +1,64 @@
-        self.relax = kwargs.get('relax', 0.1)
-        self.maxiter = kwargs.get('maxiter', 100000)
-        self.eps = kwargs.get('eps', 1e-4)
-
-    def unconstrained_bestfit(self, objective, data, pdf, init_pars, par_bounds):
-        # the graph
-        data = self.tb.astensor(data)
-        parlist = [self.tb.astensor([p]) for p in init_pars]
-
-        pars = self.tb.concatenate(parlist)
-        objective = objective(pars, data, pdf)
-        hessian = tf.hessians(objective, pars)[0]
-        gradient = tf.gradients(objective, pars)[0]
-        invhess = tf.linalg.inv(hessian)
-        update = tf.transpose(tf.matmul(invhess, tf.transpose(tf.stack([gradient]))))[0]
-
-        # run newton's method
-        best_fit = init_pars
-        for i in range(self.maxiter):
-            up = self.tb.session.run(update, feed_dict={pars: best_fit})
-            best_fit = best_fit - self.relax * up
-            if np.abs(np.max(up)) < self.eps:
-                break
+    return func
 
-        return best_fit.tolist()
 
-    def constrained_bestfit(
-        self, objective, constrained_mu, data, pdf, init_pars, par_bounds
-    ):
-        # the graph
-        data = self.tb.astensor(data)
+class tflow_optimizer(AutoDiffOptimizerMixin):
+    def __init__(self, *args, **kargs):
+        pass
 
-        nuis_pars = [
-            self.tb.astensor([p])
-            for i, p in enumerate(init_pars)
-            if i != pdf.config.poi_index
-        ]
-        poi_par = self.tb.astensor([constrained_mu])
+    def setup_unconstrained(self, objective, pdf, data, init_pars, par_bounds):
+        pars_placeholder = tf.placeholder(tf.float32, (pdf.config.npars,))
+        data_placeholder = tf.placeholder(
+            tf.float32, (pdf.config.nmaindata + pdf.config.nauxdata,)
+        )
+        nll = objective(pars_placeholder, data_placeholder, pdf)
+        nllgrad = tf.identity(tf.gradients(nll, pars_placeholder)[0])
+        func = eval_func([nll, nllgrad], pars_placeholder, data_placeholder, data)
+        return func, init_pars, par_bounds
 
-        nuis_cat = self.tb.concatenate(nuis_pars)
-        pars = self.tb.concatenate(
-            [
-                nuis_cat[: pdf.config.poi_index],
-                poi_par,
-                nuis_cat[pdf.config.poi_index :],
-            ]
+    def setup_constrained(self, objective, poival, pdf, data, init_pars, par_bounds):
+        tensorlib, _ = get_backend()
+        data_placeholder = tf.placeholder(
+            tf.float32, (pdf.config.nmaindata + pdf.config.nauxdata,)
         )
-        objective = objective(pars, data, pdf)
-        hessian = tf.hessians(objective, nuis_cat)[0]
-        gradient = tf.gradients(objective, nuis_cat)[0]
-        invhess = tf.linalg.inv(hessian)
-        update = tf.transpose(tf.matmul(invhess, tf.transpose(tf.stack([gradient]))))[0]
+        idx = default_backend.astensor(range(pdf.config.npars), dtype='int')
+        init = default_backend.astensor(init_pars)
```

Comment:
See question above RE: if the `astensor` calls that don't explicitly transform the `dtype` are needed as a result of use in `scipy.optimize.minimize`.